pay-api/src/pay_api/models/custom_query.py

```diff
@@ -12,20 +12,26 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Custom Query class to extend BaseQuery class functionality."""
-from datetime import datetime
+from datetime import date, datetime
 from flask_sqlalchemy import BaseQuery
 from sqlalchemy import func
 
 
 class CustomQuery(BaseQuery):
     """Custom Query class to extend the base query class for helper functionality."""
 
-    def filter_conditionally(self, search_criteria, model_attribute):
+    def filter_conditionally(self, search_criteria, model_attribute, is_like: bool = False):
```

Comment:
only suggestion I have here, maybe an enum changing the type of filtering? that way it's more explicit?

`DATE, DATETIME, IS_LIKE`

Reply:
I think this is good the way it is actually